### ISSUE TITLE
Add option to "powerlevel" troops

### ISSUE THREAD

**Issue by cpuritscher** [TLDRMissions/TLDRMissions#258]
**TLDR Version of your idea**
TLDR version: Add the option to powerlevel the first half of your highest followers + 1 to quickly increase the troop level (which uses the median level of your followers).

**Slightly longer version**
Similar to the special XP behaviour it would be great of we had another option, that focuses the XP missions on your highest half +1 of your followers only.

As a bonus it would be great if there was another button (or does it automatically after the missions have finished) that applies the (accountwide) XP token items on your followers (although I don't even know if there is a way to automate that).

**Comment by teelolws** [comment]
I see what you mean about working on the median level follower. I'll think about it.

> As a bonus it would be great if there was another button (or does it automatically after the missions have finished) that applies the (accountwide) XP token items on your followers (although I don't even know if there is a way to automate that).

This is a bit outside the scope of my addon. Someone else can make an addon that does this. Might even be able to just stuff it into a macro.

**Comment by teelolws** [comment]
Okay heres my work in progress so far.

First, the option. A dropdown to change the algorithm that "Special treatment" will use. The default will remain as it was before - just level all the troops. Option 3 is what you're asking for:
![image](https://user-images.githubusercontent.com/49792789/170169493-edf02712-c5f5-47c6-93c5-03c83a4cbfdc.png)

Now, in this example, I have 12 followers. 5 of them are level 47. The rest are level 46. To optimize levelling the troops, it is going to put 2 level 46 followers in, and 3 level 47 followers. The level 46 followers are decided by their "garrFollowerID", which is a number coded in by Blizzard. This way, it will keep choosing those same two over the other level 46s even tomorrow.
![image](https://user-images.githubusercontent.com/49792789/170170154-33440c19-c064-470a-b616-b626cf84716b.png)

I'm going to do more testing before I release this. I expect this to go in tomorrows patch.

**Comment by cpuritscher** [comment]
That sounds great so far, although I don't understand why you would sort and filter them by their internal ID and not simply take the 7 followers with the highest amount of XP, and for the actual XP mission you send the 5 (available) followers with the lowest XP from the highest 7. 
This way you still would always send the same characters, unless there is an actual change of XP rank order (for example by manually applying a XP item) - but then it's also correct that the prioritized followers change.

**Comment by teelolws** [comment]
That would make more sense, yes.
